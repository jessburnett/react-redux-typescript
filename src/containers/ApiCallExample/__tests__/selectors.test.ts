import { fromJS } from 'immutable';

import {
  selectData,
  selectError,
  selectLoading,
} from "../selectors";

describe('selectors', () => {

  let state;

  beforeEach(() => {
    state = fromJS({
      apiCallExample: {
        data: 'test data',
        loading: false,
        error: 'test error',
      },
    });
  });

  it('correctly select data', () => {
    expect(selectData(state)).toMatchSnapshot();
  });

  it('correctly select error', () => {
    expect(selectError(state)).toMatchSnapshot();
  });

  it('correctly selects loading', () => {
    expect(selectLoading(state)).toMatchSnapshot();
  });

});
