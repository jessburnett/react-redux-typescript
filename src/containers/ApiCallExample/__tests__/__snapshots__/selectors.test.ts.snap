exports[`selectors correctly select data 1`] = `"test data"`;

exports[`selectors correctly select error 1`] = `"test error"`;

exports[`selectors correctly selects loading 1`] = `false`;
